src/GameQ/Protocols/Eos.php

```diff
@@ -0,0 +1,275 @@
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GameQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace GameQ\Protocols;
+
+use GameQ\Exception\Protocol as Exception;
+use GameQ\Server;
+
+/**
+ * Epic Online Services Protocol Class
+ *
+ * Serves as a base class for EOS-powered games.
+ *
+ * @package GameQ\Protocols
+ * @author  H.Rouatbi
+ */
+class Eos extends Http
+{
+    /**
+     * The protocol being used
+     *
+     * @var string
+     */
+    protected $protocol = 'eos';
+
+    /**
+     * Longer string name of this protocol class
+     *
+     * @var string
+     */
+    protected $name_long = 'Epic Online Services';
+
+    /**
+     * String name of this protocol class
+     *
+     * @var string
+     */
+    protected $name = 'eos';
+
+    /**
+     * Grant type used for authentication
+     *
+     * @var string
+     */
+    protected $grant_type = 'client_credentials';
+
+    /**
+     * Deployment ID for the game or application
+     *
+     * @var string
+     */
+    protected $deployment_id = null;
+
+    /**
+     * User ID for authentication
+     *
+     * @var string
+     */
+    protected $user_id = null;
+
+    /**
+     * User secret key for authentication
+     *
+     * @var string
+     */
+    protected $user_secret = null;
+
+    /**
+     * Holds the server ip so we can overwrite it back
+     *
+     * @var string
+     */
+    protected $serverIp = null;
+
+    /**
+     * Holds the server port query so we can overwrite it back
+     *
+     * @var string
+     */
+    protected $serverPortQuery = null;
+
+    /**
+     * Normalize some items
+     *
+     * @var array
+     */
+    protected $normalize = [
+        // General
+        'general' => [
+            // target       => source
+            'hostname'   => 'hostname',
+            'mapname'    => 'mapname',
+            'maxplayers' => 'maxplayers',
+            'numplayers' => 'numplayers',
+            'password'   => 'password',
+        ]
+    ];
+
+    /**
+     * Process the response from the EOS API
+     *
+     * @return array
+     * @throws Exception
+     */
+    public function processResponse()
+    {
+        // Authenticate and get the access token
+        $auth_token = $this->authenticate();
+
+        // If authentication failed, throw an exception
+        if (!$auth_token) {
+            throw new Exception('Failed to authenticate with EOS API.');
+        }
+
+        // Query for server data
+        $server_data = $this->queryServers($auth_token);
```

Comment:
This has to be solved throught the Protocol itself, the process Response method is intended to really only process the response, not do any queries. Please look at the main review note for more information.